Test Failure: MigrationTest

### Describe the bug

Tests **migration2_xTest** and **migration9_xTest** are failing.

### Version

17.0.0

### Expected behavior

Tests should pass successfully.

### Actual behavior

Tests are failing in the following environments:
**OS:** RHEL (any Linux should work)
**Test group:** DB Migration and DB Migration - 2.5.5
**DB:** postgresql132 / postgresplus131 / mysql80 / mariadb103 / mssql2019 / mariadb_galera_103 / oracle19cRAC

**Stack trace:**
`java.lang.AssertionError: Error when refreshing offline token. Error: unknown_error, Error details: null, offline token from previous version: eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJCMHFMbWlvRkx3VzZMV0RIMllIRVR0TzF2alh5QVQtUXM5blpoYlBFU3pJIn0.eyJqdGkiOiIwZDA2M2Q2MS1mYTgxLTRmNzYtYTA0YS1iNzMxYjE5ZjEyODIiLCJleHAiOjAsIm5iZiI6MCwiaWF0IjoxNjQyMDIzNTg2LCJpc3MiOiJodHRwOi8vbG9jYWxob3N0OjgxODAvYXV0aC9yZWFsbXMvTWlncmF0aW9uIiwiYXVkIjoibWlncmF0aW9uLXRlc3QtY2xpZW50Iiwic3ViIjoiNTU2ZWI0MzAtZDU3NC00OTU2LTkwOGEtODM1MjdhNzc5MzJhIiwidHlwIjoiT2ZmbGluZSIsImF6cCI6Im1pZ3JhdGlvbi10ZXN0LWNsaWVudCIsImF1dGhfdGltZSI6MCwic2Vzc2lvbl9zdGF0ZSI6IjE0MTU5M2JkLWUwMTgtNDBiNy05YTVhLTczNzcxMTNmNDA5NyIsImNsaWVudF9zZXNzaW9uIjoiYzYxYTczNTgtYzMzOS00N2RjLThkMjMtOTk0NzAxMjY2YjgyIiwicmVhbG1fYWNjZXNzIjp7InJvbGVzIjpbIm9mZmxpbmVfYWNjZXNzIiwidW1hX2F1dGhvcml6YXRpb24iXX0sInJlc291cmNlX2FjY2VzcyI6eyJhY2NvdW50Ijp7InJvbGVzIjpbIm1hbmFnZS1hY2NvdW50Iiwidmlldy1wcm9maWxlIl19fX0.gKB3ymmocBEfqMJdNRuB191_Etw3SW0-lDxDj9pzoT7hdiaNphGgH9ZzN93ZnSMS4vzS_EamKxAFJ8OIY-EFX9Ay67AqS48PKVjqvgQIrNqugQLaPGVT0ZypZ9T_uyZpvDPHGhEdK9zJ9lZXg-Q0DYZy6QyeWxWIeVdjVsvU1x_HxhtfJYDwLGAgglxIMP_0ekSArowE5HBdOGJ8aIflMcgoSZp7hcA6V57vwkTWqgnKixQxk0_W--AaGxfLvqvTKT-RkGmokMBFDwEYyjphmTG4hgAljLqUP8ngZ_VIDuY8Gf6vEzYSKhecOj_WNSIKG_C-o2iB6RGBvX2e_IWcvw
	at org.junit.Assert.fail(Assert.java:89)
	at org.keycloak.testsuite.migration.AbstractMigrationTest.testOfflineTokenLogin(AbstractMigrationTest.java:693)
	at org.keycloak.testsuite.migration.MigrationTest.migration2_xTest(MigrationTest.java:127)
`

### How to Reproduce?

Execute the tests in an environment like the one described above.

### Anything else?

https://keycloak-jenkins/job/universal-test-pipeline-server/1163/testReport/

--- a/services/src/main/java/org/keycloak/authentication/AuthenticatorUtil.java
+++ b/services/src/main/java/org/keycloak/authentication/AuthenticatorUtil.java
@@ -17,8 +17,10 @@
 
 package org.keycloak.authentication;
 
+import org.keycloak.models.AuthenticatedClientSessionModel;
 import org.keycloak.models.Constants;
 import org.keycloak.sessions.AuthenticationSessionModel;
+import org.keycloak.sessions.CommonClientSessionModel;
 
 public class AuthenticatorUtil {
 
@@ -39,5 +41,10 @@ public class AuthenticatorUtil {
     public static boolean isLevelOfAuthenticationSatisfied(AuthenticationSessionModel authSession) {
         return AuthenticatorUtil.getRequestedLevelOfAuthentication(authSession)
             <= AuthenticatorUtil.getCurrentLevelOfAuthentication(authSession);
+    }
+
+    public static int getCurrentLevelOfAuthentication(AuthenticatedClientSessionModel clientSession) {
+        String clientSessionLoaNote = clientSession.getNote(Constants.LEVEL_OF_AUTHENTICATION);
+        return clientSessionLoaNote == null ? Constants.NO_LOA : Integer.parseInt(clientSessionLoaNote);
     }
 }

--- a/services/src/main/java/org/keycloak/protocol/oidc/TokenManager.java
+++ b/services/src/main/java/org/keycloak/protocol/oidc/TokenManager.java
@@ -854,7 +854,7 @@ public class TokenManager {
     }
 
     private String getAcr(AuthenticatedClientSessionModel clientSession) {
-        int loa = Integer.parseInt(clientSession.getNote(Constants.LEVEL_OF_AUTHENTICATION));
+        int loa = AuthenticatorUtil.getCurrentLevelOfAuthentication(clientSession);
         if (loa < Constants.MINIMUM_LOA) {
             loa = AuthenticationManager.isSSOAuthentication(clientSession) ? 0 : 1;
         }

